src/test/kotlin/org/openrewrite/java/migrate/lombok/LombokValToFinalVarTest.kt

```diff
@@ -0,0 +1,146 @@
+/*
+ * Copyright 2021 the original author or authors.
+ * <p>
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * <p>
+ * https://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.openrewrite.java.migrate.lombok
+
+import org.junit.jupiter.api.Test
+import org.openrewrite.Issue
+import org.openrewrite.java.Assertions.*
+import org.openrewrite.java.JavaParser
+import org.openrewrite.test.RecipeSpec
+import org.openrewrite.test.RewriteTest
+
+class LombokValToFinalVarTest : RewriteTest {
+    override fun defaults(spec: RecipeSpec) {
+        spec.recipe(LombokValToFinalVar())
+        spec.parser(JavaParser.fromJavaVersion().classpath("lombok"))
+    }
+
+    @Test
+    fun replaceAssignment() = rewriteRun(
+        version(
+            java(
+                """
+                import lombok.val;
+                class A {
+                    void bar() {
+                        val foo = "foo";
+                    }
+                }
+                """,
+                """
+                class A {
+                    void bar() {
+                        final var foo = "foo";
+                    }
+                }
+                """),
+                17))
+
+    @Test
+    fun retainPrefixComment() = rewriteRun(
+        version(
+            java(
+                """
+                import lombok.val;
+                class A {
+                    void bar() {
+                        // Prefix
+                        val foo = "foo";
+                    }
+                }
+                """,
+                """
+                class A {
+                    void bar() {
+                        // Prefix
+                        final var foo = "foo";
+                    }
+                }
+                """),
+                17))
+
+    @Test
+    fun retainInfixComment() = rewriteRun(
+        version(
+            java(
+                """
+                import lombok.val;
+                class A {
+                    void bar() {
+                        val foo = 
+                        // Infix
+                        "foo";
+                    }
+                }
+                """,
+                """
+                class A {
+                    void bar() {
+                        final var foo = 
+                        // Infix
+                        "foo";
```

Comment:
One could argue that both `retainInfixComment` & `retainWhitespace` should be rare, and therefore not worth supporting. That would be fine by me, but thought to make that decision explicit here, to make clear it's not an oversight if we drop it (for) now.

Reply:
My thought is that even though `NamedVariable` comments are rare they exist for a reason and they should be accounted for.  In this case the prefix from the original variable can just be copied to the transformed variable.

check this out:
```java
J.VariableDeclarations.NamedVariable nv = mv.getVariables().get(0);
return varDecls.withVariables(ListUtils.map(varDecls.getVariables(),
    varDecl -> varDecl.withInitializer(varDecl.getInitializer().withPrefix(nv.getInitializer().getPrefix()))));
```